Automattic/pocket-casts-android
# Playback 2023: Entry point

| 📘 Part of: #1463 | 🎨 Designs: lH66LwxxgG8btQ8NrM0ldx-fi-1599_20385 |
|:---:|:---:|

## Description
Updates the logic and visuals for the Playback 2023 modal, profile badge, and profile card.

## Testing Instructions
1. Start the app with an account that has a few episodes listened
2. ✅ A modal should be shown
3. ✅  Tapping on "View My 2023" shows the stories
4. Dismiss it
5. ✅ There should be a badge on the "Profile" tab
6. Tap it
7. ✅ The badge should disappear
8. ✅ Check that there's a card on Profile prompting you to see your 2023
9. Tap it
10. ✅ Playback 2023 stories should start

If you want to reset the modal being shown and badge, go to Profile > Settings > Developer > Reset modal/profile badge

## Screenshots or Screencast 

Profile Entry | Profile Dot | Modal Entry
----|----|----
<img width=300 src="https://github.com/Automattic/pocket-casts-android/assets/1405144/ef546786-5119-4e3f-82cf-9de99c5c80a2"/> | <img width=300 src="https://github.com/Automattic/pocket-casts-android/assets/1405144/83700dc4-f1b7-4a2d-b092-5c6aca0560fe"/> | <img width=300 src="https://github.com/Automattic/pocket-casts-android/assets/1405144/e28f1963-04d3-4f8f-860b-30bdd26f2fc3"/>


Tablet Profile Entry | Tablet Modal Entry 
-------|----
![tablet_lt](https://github.com/Automattic/pocket-casts-android/assets/1405144/410a4c3b-db84-407f-b68e-29a0d2f321f8) | ![tablet](https://github.com/Automattic/pocket-casts-android/assets/1405144/75d4cffa-9a71-4898-8e8b-afae5cdc3774) 


## Checklist
 
#### I have tested any UI changes...
